src/coreclr/vm/callhelpers.cpp

```diff
@@ -234,18 +238,22 @@ void * DispatchCallSimple(
     callDescrData.fpReturnSize = 0;
     callDescrData.pTarget = pTargetAddress;
 
-    if ((dwDispatchCallSimpleFlags & DispatchCallSimple_CatchHandlerFoundNotification) != 0)
-    {
-        DispatchCallDebuggerWrapper(
-            &callDescrData,
-            dwDispatchCallSimpleFlags & DispatchCallSimple_CriticalCall);
-    }
-    else
-    {
-        CallDescrWorkerWithHandler(&callDescrData, dwDispatchCallSimpleFlags & DispatchCallSimple_CriticalCall);
-    }
-
-    return *(void **)(&callDescrData.returnValue);
+    ARG_SLOT result;
```

Comment:
A note for future - this will not be sufficient e.g. for x64 Unix ABI that can return result in two general purpose registers or two xmm registers.